src/VisualStudio/TestUtilities/Remoting/WorkspaceWrapper.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using RoslynProject = Microsoft.CodeAnalysis.Project;
+using RoslynWorkspace = Microsoft.CodeAnalysis.Workspace;
+
+namespace Roslyn.VisualStudio.Test.Utilities.Remoting
+{
+    internal class WorkspaceWrapper : MarshalByRefObject
+    {
+        private RoslynWorkspace _workspace;
+
+        public WorkspaceWrapper(RoslynWorkspace workspace)
+        {
+            _workspace = workspace;
+        }
+
+        public void AddReference(string projectName, string referenceAssemblyPath)
+        {
+            var metadataReference = MetadataReference.CreateFromFile(referenceAssemblyPath);
+            RoslynProject newProject = GetProject(projectName).AddMetadataReference(metadataReference);
+            _workspace.TryApplyChanges(newProject.Solution);
+        }
+
+        public void OpenDocument(string projectName, string documentName)
+        {
+            RoslynProject project = GetProject(projectName);
+            DocumentId documentId = project.Documents.FirstOrDefault(doc => doc.Name.Equals(documentName)).Id;
+
+            if (documentId != null)
```

Comment:
I think we should throw an exception if `documentId == null`, otherwise the call to `OpenDocument` could do nothing and the caller could end up modifying an unexpected document.